app/tests/unit/routes/v1/webade.spec.js

```diff
@@ -0,0 +1,448 @@
+    expect(webadeSpy).toHaveBeenCalledTimes(1);
+    expect(webadeSpy).toHaveBeenCalledWith(body.configForm, body.passwordPublicKey, keycloakAuthMock.grant.access_token.content.sub);
+  });
+});
+
+
+describe(`GET ${basePath}/:webAdeEnv/:appAcronym/dependencies`, () => {
+  const permissionSpy = jest.spyOn(permissionHelpers, 'checkAcronymPermission');
+  const utilsSpy = jest.spyOn(utils, 'filterWebAdeDependencies');
+  const webadeSpy = jest.spyOn(webade, 'getAppConfigs');
+
+  beforeEach(() => {
+    permissionSpy.mockReset();
+    utilsSpy.mockReset();
+    webadeSpy.mockReset();
+  });
+
+  it('should yield a 422 if param validation fails', async () => {
+    const response = await request(app).get(`${basePath}/DEV/XXX/dependencies`);
+
+    expect(response.statusCode).toBe(422);
+    expect(response.body).toBeTruthy();
+    expect(response.body.detail).toMatch('Validation failed');
+    expect(webadeSpy).toHaveBeenCalledTimes(0);
+  });
+
+  it('should yield a 403 if user has no WEBADE_CFG_READ_ALL and the permission check fails', async () => {
+    const clone = JSON.parse(JSON.stringify(keycloakAuthMock));
+    clone.grant.access_token.content.realm_access.roles = ['SOME_ROLE'];
+    const overrideApp = helper.expressHelper(basePath, router, clone);
+
+    permissionSpy.mockResolvedValue('error');
+
+    const response = await request(overrideApp).get(`${basePath}/INT/XXX/dependencies`);
+
+    expect(response.statusCode).toBe(403);
+    expect(response.body).toBeTruthy();
+    expect(response.body.detail).toMatch('error');
+    expect(permissionSpy).toHaveBeenCalledTimes(1);
+    expect(webadeSpy).toHaveBeenCalledTimes(0);
+  });
+
+  it('should return a valid response if user has no WEBADE_CFG_READ_ALL and the permission check passes', async () => {
+    const clone = JSON.parse(JSON.stringify(keycloakAuthMock));
+    clone.grant.access_token.content.realm_access.roles = ['SOME_ROLE'];
+    const overrideApp = helper.expressHelper(basePath, router, clone);
+
+    permissionSpy.mockResolvedValue();
+    // No idea why mockResolvedValue doesn't work on this method??? But have to leave for now
+    //utilsSpy.mockResolvedValue({ test: '123' });
+    utilsSpy.mockImplementation(() => {
+      return { test: '123' };
+    });
+    webadeSpy.mockResolvedValue({ test: '123', more: 567 });
+
+    const response = await request(overrideApp).get(`${basePath}/INT/XXX/dependencies`);
+
+    expect(response.statusCode).toBe(200);
+    expect(response.body).toBeTruthy();
+    expect(response.body).toEqual({ test: '123' });
+    expect(permissionSpy).toHaveBeenCalledTimes(1);
+    expect(utilsSpy).toHaveBeenCalledTimes(1);
+    expect(webadeSpy).toHaveBeenCalledTimes(1);
+  });
+
+  it('should return a valid response if user has WEBADE_CFG_READ_ALL', async () => {
+    // No idea why mockResolvedValue doesn't work on this method??? But have to leave for now
+    //utilsSpy.mockResolvedValue({ test: '123' });
+    utilsSpy.mockImplementation(() => {
```

Comment:
If this isn't an async function, you should be using mockReturnValue instead of mockResolvedValue/mockRejectedValue

Reply:
Thanks!!